src/dxvk/dxvk_util.cpp

```diff
@@ -142,6 +142,90 @@ namespace dxvk::util {
   }
 
 
+  void packImageData(
+          void*             dstBytes,
+    const void*             srcBytes,
+          VkDeviceSize      srcRowPitch,
+          VkDeviceSize      srcSlicePitch,
+          VkDeviceSize      dstRowPitch,
+          VkDeviceSize      dstSlicePitch,
+          VkImageType       imageType,
+          VkExtent3D        imageExtent,
+          uint32_t          imageLayers,
+    const DxvkFormatInfo*   formatInfo,
+          VkImageAspectFlags aspectMask) {
+    for (uint32_t i = 0; i < imageLayers; i++) {
+      auto dstData = reinterpret_cast<      char*>(dstBytes);
+      auto srcData = reinterpret_cast<const char*>(srcBytes);
+
+      for (auto aspects = aspectMask; aspects; ) {
+        auto aspect = vk::getNextAspect(aspects);
+        auto extent = imageExtent;
+        auto elementSize = formatInfo->elementSize;
+
+        if (formatInfo->flags.test(DxvkFormatFlag::MultiPlane)) {
+          auto plane = &formatInfo->planes[vk::getPlaneIndex(aspect)];
+          extent.width  /= plane->blockSize.width;
+          extent.height /= plane->blockSize.height;
+          elementSize = plane->elementSize;
+        }
+
+        auto blockCount = computeBlockCount(extent, formatInfo->blockSize);
+
+        VkDeviceSize bytesTotal = blockCount.depth * std::min(dstRowPitch, srcRowPitch);
+
+        const bool directCopy = ((dstRowPitch   == srcRowPitch  ) || (blockCount.height == 1))
+                             && ((dstSlicePitch == srcSlicePitch) || (blockCount.depth  == 1));
```

Comment:
this is wrong, need to check if both src and dst pitches are equal to the byte count.